[Feature] Add prev/next/back links to pool candidate page

esizer opened the pull request:
🤖 Resolves #9017 

## 👋 Introduction

This adds the prev/next/back links to the candidate page when navigating from the assessment tracker.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build `npm run dev`
2. Login as `admin@test.com`
3. Navigate to a pool's screening and assessment page 
4. Take note of the order candidates appear in a column
5. Click on one
6. Confirm the next/previous links appear
7. Step through confirming that the links navigate to the expected candidate
8. Navigate to the start of the column (first candidate)
9. Confirm the previous link does not appear
10. Navigate to the last candidate
11. Confirm the next link does not appear
12. Confirm that the return to assessment tracker link appears
13. Navigate to the pool candidate from anywhere but the assessment tracker
    - or, empty cache and hard reload
14. Confirm the navigation does not exist

## 📸 Screenshot

### First candidate
![Screenshot 2024-02-09 135610](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/692d479f-eceb-4b73-9fd0-b35b968bf8c8)

### Middle of the List
![Screenshot 2024-02-09 135341](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/c1fd8534-1d7f-46b8-b84e-db5923029db5)

### Last candidate
![Screenshot 2024-02-09 135355](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/eef35dbe-a0ea-4bf0-ae34-a6449287753c)

mnigh commented on apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/RODViewPoolCandidatePage.tsx:
```diff
@@ -646,20 +646,10 @@ export const ViewPoolCandidate = ({
                 })}
               </p>
               <p>Step 1: Screening Application (Replace with fetched value)</p>
```
Is **Replace with fetched value** meant to remain?

## Screenshot
<img width="304" alt="Screen Shot 2024-02-12 at 09 39 13" src="https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/762b0668-2197-4339-9be8-c23cd17c42d5">

esizer replied:
My guess is that it will be similar to [the pill we recently added](https://github.com/GCTC-NTGC/gc-digital-talent/issues/9146) but include the step name rather than the status 🤔 

@Jerryescandon is that accurate?

mnigh replied:
Ok, will this be removed as part of another PR then? I just want to make sure this is out of scope for this issue.

mnigh reviewed (commented):
Works according to the testing steps, however the navigation is awkward to access on screens with a shorter height as I needed to scroll the entire page to see the links.

## Screen recording
https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/99bd817c-9859-4692-91a3-868b668fd40b

mnigh reviewed (changes requested):
I found a bug where when switching language that the `CandidateNavigation` component is not rendered after switching languages.

## Screen recording

https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/1e841109-f5a9-4e83-98aa-253d45f846df

The **Back to assessments tracker** link still gets partially cut off but the next and previous links are improved.

## Screenshots

<img width="338" alt="Screen Shot 2024-02-15 at 10 13 11" src="https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/43afd843-b160-4134-9a4c-dcd4b548f7f4">
<img width="298" alt="Screen Shot 2024-02-15 at 10 12 59" src="https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/34d5bdeb-8e10-4f99-a08e-3ccbfecdfc40">
<img width="323" alt="Screen Shot 2024-02-15 at 10 01 41" src="https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/30ae2872-568f-454d-bb76-1e2f96b2b1ed">
<img width="299" alt="Screen Shot 2024-02-15 at 10 01 29" src="https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/cc02f4bd-67e5-41e9-80a1-c75ee4163919">

esizer commented:
> > I found a bug where when switching language that the `CandidateNavigation` component is not rendered after switching languages.
> 
> Ohhh, okay. Huh, that is going to be super, super tricky without search params since our language toggle needs to be a plain link 😖 I'll ask in stand up about this.

Okay @mnigh based on discussion at standup, we have split off state persistence to another ticket (https://github.com/GCTC-NTGC/gc-digital-talent/issues/9376). I also added the current step to the navigation in [add step name to candidate nav](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9302/commits/b74120e8c16b02c36f59f67f2997d9cd9484629c)

mnigh reviewed (approved):
This PR appears to meet the acceptance criteria from the linked issue.

I can't tell if the `stepName` actually changes from N/A ever since I have no way of changing the steps, but it does show that. The translation for N/A here is S.O. which means [Not applicable](https://talent-cloud.slack.com/archives/D03DTK11204/p1708017412013099?thread_ts=1708015889.365649&cid=D03DTK11204) which seems like it should be Not available which would be n.d. but that is just another reason to fix #8898.

## Screenshots
<img width="280" alt="Screen Shot 2024-02-15 at 12 14 51" src="https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/86b637fe-8ca2-472a-85e5-8590125bb660">
<img width="265" alt="Screen Shot 2024-02-15 at 12 15 17" src="https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/397d4a26-f107-43f2-8fab-5478109813cb">
